Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3184,42 +3163,11 @@ - (void)attachmentButtonPressed
     [self presentViewController:actionSheetController animated:true completion:nil];
 }
 
-- (NSIndexPath *)lastVisibleIndexPath
-{
-    NSIndexPath *lastVisibleIndexPath = nil;
-    for (NSIndexPath *indexPath in [self.collectionView indexPathsForVisibleItems]) {
-        if (!lastVisibleIndexPath || indexPath.row > lastVisibleIndexPath.row) {
-            lastVisibleIndexPath = indexPath;
-        }
-    }
-    return lastVisibleIndexPath;
-}
-
-- (nullable TSInteraction *)lastVisibleInteraction
-{
-    NSIndexPath *lastVisibleIndexPath = [self lastVisibleIndexPath];
-    if (!lastVisibleIndexPath) {
-        return nil;
-    }
-    return [self interactionAtIndexPath:lastVisibleIndexPath];
-}
-
-// TODO: Is this safe?
-// TODO: Remove most usages of this.
-- (TSInteraction *)interactionAtIndexPath:(NSIndexPath *)indexPath
-{
-    OWSAssert(indexPath);
-    OWSAssert(indexPath.section == 0);
-
-    ConversationViewItem *_Nullable viewItem = [self viewItemForIndex:(NSUInteger)indexPath.row];
-    return viewItem.interaction;
-}
-
 - (void)updateLastVisibleTimestamp
 {
-    TSInteraction *lastVisibleInteraction = [self lastVisibleInteraction];
-    if (lastVisibleInteraction) {
-        uint64_t lastVisibleTimestamp = lastVisibleInteraction.timestampForSorting;
+    ConversationViewItem *_Nullable lastViewItem = [self.viewItems lastObject];
```

Comment:
How do we know that the `self.viewItems.lastObject` is visible?

Reply:
Good catch, this is wrong. Opening a new PR.

Sorry, I missed your comments last night.